### ISSUE TITLE
More options for nearest neighbor selection for KdTreeCrossmatch

### ISSUE THREAD

**Issue by hombit** [astronomy-commons/lsdb#151]
Currently, we have two ways to control neighbor selection with the KdTreeCrossmatch algorithm: the maximum number of neighbors to select `n_neighbors` and the maximum radius `radius_arcsec`. However, it would be beneficial to add more flexibility:

- [ ] Set a minimum radius and/or skip the first nth matches. This would be very useful for catalog self-matching tasks.
- [ ] Support a "left" join in addition to the "inner" join we have now, so that rows from the left table that do not match are kept in the result.

I propose that we keep this issue open for discussions while creating separate issues for individual tasks.

**Comment by nevencaplar** [comment]
@hombit Can you just explain a little bit what you mean by ``set a minimum radius'' - skip a match if it it below a certain distance? Is this so that you ensure that you are matching to a different object in a crossmatch, making sure to avoid the same object in the second catalog?

**Comment by hombit** [comment]
@nevencaplar, yes, I meant this. I was thinking about catalog self-matching. For example, you'd like to find wide binaries and looking for pairs of stars in 2''. However, you wouldn't like to have objects to match to themselves, so you would set `nthmatch=2`. However, it wouldn't work with some catalogs having the same sky source to be represented by few rows, for instance "source" catalogs, or catalogs with duplicated objects such as PS1 DR2.

**Comment by nevencaplar** [comment]
What is happening here? I see that this ticket has a pull from April? @camposandro ?

**Comment by camposandro** [comment]
We left #271 on hold during the feature freeze leading to the IVOA demos and I didn't have the chance to look back into it. I need to resolve the merge conflicts and request a new review.
